### DISCUSSION TITLE
Insert with autoIncrement() id error

### DISCUSSION THREAD

**Discussion by mytc** [simolus3/drift#1215] [Q&A] [answered]
I have a table like this:
```
class Metas extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get key => text()();
  TextColumn get value => text()();
}
```
I am using DOA to insert the data
 `Future insertMeta(Insertable<Meta> mt) => into(metas).insert(mt);`

Problem when I try to insert with it:
```
var meta = Meta(
id:0,
  key:'key 1',
 value:'meta value',
);
doa.InsertMeta(meta );
```
I got this:
`SqliteException (SqliteException(1555): UNIQUE constraint failed: metas.id, constraint failed (code 1555))`

The reason seem to be the `id` value, but since it is autoincrement, and `required` at the same time, there is no way I can initial the class with null. 

Does it mean, I can't use DOA to insert data? Instead, I should write own query statement for it?

**Comment by simolus3** [comment, marked as answer]
Generated data classes are meant to hold a full row in a table. For inserts, you may only need partial data (e.g. to skip the columns with default values). Moor generates companion classes for this purpose:

```dart
doa.InsertMeta(MetaCompanion.insert(key: 'key 1', value: 'meta value'));
```
